components/service/glean/src/main/java/mozilla/components/service/glean/storages/GenericScalarStorageEngine.kt

```diff
@@ -167,11 +181,30 @@ abstract class GenericScalarStorageEngine<ScalarType> : StorageEngine {
 
     /**
      * Helper function for the derived classes. It can be used to record
-     * simple scalars to the internal storage.
+     * simple scalars to the internal storage. Internally, this calls [recordScalar]
+     * with a custom `combine` function that only sets the new value.
+     *
+     * @param metricData the information about the metric
+     * @param value the new value
      */
     protected fun recordScalar(
         metricData: CommonMetricData,
         value: ScalarType
+    ) = recordScalar(metricData, value) { _, v -> v }
+
+    /**
+     * Helper function for the derived classes. It can be used to record
+     * simple scalars to the internal storage.
+     *
+     * @param metricData the information about the metric
+     * @param value the new value
+     * @param combine an optional lambda function to combine the currently stored value and
+     *        the new one; this allows to implement new behaviours such as adding.
```

Comment:
nit: `optional` is a bit misleading since the lambda is not optional for *this* method.